hal/api/critical.h

```diff
@@ -234,6 +287,14 @@ uint16_t core_util_atomic_incr_u16(uint16_t * valuePtr, uint16_t delta);
 uint32_t core_util_atomic_incr_u32(uint32_t * valuePtr, uint32_t delta);
 
 /**
+ * Atomic increment.
+ * @param  valuePtr Target memory location being incremented.
+ * @param  delta    The amount being incremented.
+ * @return          The new incremented value.
+ */
+void *core_util_atomic_incr(void **valuePtr, unsigned delta);
```

Comment:
@0xc0170, this is incongruous with the mbed style guide ([here](https://developer.mbed.org/teams/SDK-Development/wiki/mbed-sdk-coding-style)). Should the style guide be updated?
> For pointers, '*' is adjacent to a name (analogin_t *obj) 

Additionally the rest of this file uses this style `uint8_t * valuePtr`. Should I consolidate both to the style you are asking for `uint8_t* valuePtr`?

@rgrover, good point. I didn't think after copying the u32 function. I can change it to `ptrdiff_t` next chance I get ([prior art](http://en.cppreference.com/w/cpp/atomic/atomic/operator_arith2)).

I could add `PtrToVoid_t` although it does not seem useful to me here. @0xc0170?

We are not using templates here due to C support. This file is actually primarily used by C files. Actually, we are getting several warnings due to the lack of void in the C function declarations. If I'm going to touch more of this file I'll remove those.